[libnoise] LNK2019 error with libnoise

**Describe the bug**
As described on StackOverflow : https://stackoverflow.com/questions/64441733/lnk2019-error-when-using-a-library-installed-with-vcpkg
`vcpkg` is integrated to Visual Studio 2019 and works fine with any other library, but libnoise doesn't work right away. I have installed the triplet `x64-windows`, and try to compile this C++ program in x64 : 
```C++
#include <noise/noise.h> // The header is found as well as the definition of noise::module::Perlin

int main()
{
   noise::module::Perlin noise; // The implementation is not found
}
```
I get this error : 
```
1>------ Build started: Project: Mayak, Configuration: Debug x64 ------
1>main.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl noise::module::Perlin::Perlin(void)" (__imp_??0Perlin@module@noise@@QEAA@XZ) referenced in function main
1>NoiseVideoGenerator.obj : error LNK2001: unresolved external symbol "__declspec(dllimport) public: __cdecl noise::module::Perlin::Perlin(void)" (__imp_??0Perlin@module@noise@@QEAA@XZ)
1>main.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: virtual __cdecl noise::module::Perlin::~Perlin(void)" (__imp_??1Perlin@module@noise@@UEAA@XZ) referenced in function main
1>NoiseVideoGenerator.obj : error LNK2001: unresolved external symbol "__declspec(dllimport) public: virtual __cdecl noise::module::Perlin::~Perlin(void)" (__imp_??1Perlin@module@noise@@UEAA@XZ)
1>C:\Users\***\source\repos\Mayak\x64\Debug\Mayak.exe : fatal error LNK1120: 2 unresolved externals
1>Done building project "Mayak.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

**Environment**
- OS: Windows 10
- Compiler: Visual Studio 2019's compiler

**To Reproduce**
Steps to reproduce the behavior:
1. ./vcpkg install libnoise:x64-windows
2. Compile an x64 project on VS2019 using the lib (the code I pasted in the bug description for example)

**Expected behavior**
The code should compile and the library works.

**Failure logs** 
Shown in the bug description.

**Additional context**
The library works when linked manually. I add "C:\vcpkg\installed\x64-windows\bin" in "Additional Library Directories" and "noise.lib" in "Additional Dependencies".
It doesn't work when I link manually this way : "C:\vcpkg\installed\x64-windows\lib" in "Additional Library Directories" and "noise-static.lib" in "Additional Dependencies".

--- a/ports/libnoise/CONTROL
+++ b/ports/libnoise/CONTROL
@@ -1,3 +1,5 @@
 Source: libnoise
 Version: 1.0.0
+Port-Version: 1
+Homepage: https://github.com/RobertHue/libnoise
 Description: A general-purpose library that generates three-dimensional coherent noise. Useful for terrain generation and procedural texture generation. Uses a broad number of techniques (Perlin noise, ridged multifractal, etc.) and combinations of those techniques.

--- a/ports/libnoise/fix-build.patch
+++ b/ports/libnoise/fix-build.patch
@@ -1,0 +1,56 @@
+diff --git a/noiseutils/CMakeLists.txt b/noiseutils/CMakeLists.txt
+index 07747de..08c0bda 100644
+--- a/noiseutils/CMakeLists.txt
++++ b/noiseutils/CMakeLists.txt
+@@ -22,8 +22,11 @@ if(BUILD_SHARED_LIBS)
+ 	target_include_directories(${TARGET_NAME} PRIVATE ${PROJECT_SOURCE_DIR}/src)
+ 	
+ 	# install dynamic libraries (.dll or .so) into /bin
+-	install(TARGETS ${TARGET_NAME} DESTINATION "${CMAKE_INSTALL_PREFIX}/bin")
+-endif()
++	install(TARGETS ${TARGET_NAME}
++        RUNTIME DESTINATION "${CMAKE_INSTALL_PREFIX}/bin"
++        LIBRARY DESTINATION "${CMAKE_INSTALL_PREFIX}/lib"
++        ARCHIVE DESTINATION "${CMAKE_INSTALL_PREFIX}/lib")
++else()
+ 
+ #----------------------------------------
+ # build static lib (it's good practice to include a lib file for the dll)
+@@ -35,7 +38,7 @@ target_include_directories(${TARGET_NAME} PRIVATE ${PROJECT_SOURCE_DIR}/src)
+ # install static libraries (.lib) into /lib
+ install(TARGETS ${TARGET_NAME} DESTINATION "${CMAKE_INSTALL_PREFIX}/lib")
+ #----------------------------------------
+-
++endif()
+ # install include files into /include
+ install( FILES "${PROJECT_SOURCE_DIR}/noiseutils/noiseutils.h" 
+ 		DESTINATION "${CMAKE_INSTALL_PREFIX}/include/noise" )
+\ No newline at end of file
+diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
+index 2757f30..7a135c2 100644
+--- a/src/CMakeLists.txt
++++ b/src/CMakeLists.txt
+@@ -62,8 +62,11 @@ if(BUILD_SHARED_LIBS)
+ 	set_target_properties(${TARGET_NAME} PROPERTIES VERSION ${LIBNOISE_VERSION})
+ 	target_include_directories(${TARGET_NAME} PRIVATE ${PROJECT_SOURCE_DIR}/src)
+ 	target_compile_definitions(${TARGET_NAME} PRIVATE NOISE_BUILD_DLL)
+-	install(TARGETS ${TARGET_NAME} DESTINATION "${CMAKE_INSTALL_PREFIX}/bin")
+-endif()
++	install(TARGETS ${TARGET_NAME}
++        RUNTIME DESTINATION "${CMAKE_INSTALL_PREFIX}/bin"
++        LIBRARY DESTINATION "${CMAKE_INSTALL_PREFIX}/lib"
++        ARCHIVE DESTINATION "${CMAKE_INSTALL_PREFIX}/lib")
++else()
+ 
+ #----------------------------------------
+ # build static lib (it's good practice to include a lib file for the dll)
+@@ -76,7 +79,7 @@ target_compile_definitions(${TARGET_NAME} PUBLIC NOISE_STATIC)
+ # install static libraries (.lib) into /lib
+ install(TARGETS ${TARGET_NAME} DESTINATION "${CMAKE_INSTALL_PREFIX}/lib")
+ #----------------------------------------
+-
++endif()
+ # install include files into /include
+ install( DIRECTORY   "${PROJECT_SOURCE_DIR}/src/noise" 
+ 		 DESTINATION "${CMAKE_INSTALL_PREFIX}/include" )
+\ No newline at end of file

--- a/ports/libnoise/portfile.cmake
+++ b/ports/libnoise/portfile.cmake
@@ -1,5 +1,3 @@
-include(vcpkg_common_functions)
-
 set( LIBNOISE_VERSION "1.0.0" )
 set( LIBNOISE_COMMIT "d7e68784a2b24c632868506780eba336ede74ecd" )
 
@@ -9,21 +7,29 @@ vcpkg_from_github(
     REF ${LIBNOISE_COMMIT}
     SHA512 8c4d654acb4ae3d90ee62ebdf0447f876022dcb887ebfad88f39b09d29183a58e6fc1b1f1d03edff804975c8befcc6eda33c44797495285aae338c2e869a14d7
     HEAD_REF master
+    PATCHES fix-build.patch
 )
 
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA
+    OPTIONS
+        -DBUILD_WALL=ON
+        -DBUILD_SPEED_OPTIMIZED=ON
+        -DBUILD_LIBNOISE_DOCUMENTATION=OFF
+        -DBUILD_LIBNOISE_UTILS=ON
+        -DBUILD_LIBNOISE_EXAMPLES=OFF
 )
 
 vcpkg_install_cmake()
 
-# Include files should not be duplicated into the /debug/include directory.
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 
-# Handle copyright
-file(COPY ${SOURCE_PATH}/LICENSE.md DESTINATION ${CURRENT_PACKAGES_DIR}/share/libnoise)
-file(RENAME ${CURRENT_PACKAGES_DIR}/share/libnoise/LICENSE.md ${CURRENT_PACKAGES_DIR}/share/libnoise/copyright)
+if (VCPKG_LIBRARY_LINKAGE STREQUAL static)
+    vcpkg_replace_string(${CURRENT_PACKAGES_DIR}/include/noise/module/modulebase.h
+        "if NOISE_STATIC" "if 1" )
+endif()
 
-# Post-build test for cmake libraries
-vcpkg_test_cmake(PACKAGE_NAME libnoise)
+file(INSTALL ${SOURCE_PATH}/cmake/Modules/FindLibNoise.cmake DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
+
+file(INSTALL ${SOURCE_PATH}/LICENSE.md DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)

